azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/PackageMojo.java

```diff
@@ -304,8 +312,33 @@ protected void installExtension(final FunctionCoreToolsHandler handler,
     }
 
     protected boolean isInstallingExtensionNeeded(Set<BindingEnum> bindingTypes) {
-        return bindingTypes.stream().anyMatch(binding ->
+        final JsonObject hostJson = readHostJson();
+        final JsonObject extensionBundle = hostJson.getAsJsonObject(EXTENSION_BUNDLE);
+        if (extensionBundle != null &&
+                StringUtils.equalsIgnoreCase(extensionBundle.get("id").getAsString(), EXTENSION_BUNDLE_ID)) {
```

Comment:
extensionBundle.get("id") will return null if id doesn't exist?

Reply:
Yes, there may be NPE here, fixed